IPython/core/oinspect.py

```diff
@@ -870,8 +870,7 @@ def info(self, obj, oname='', formatter=None, info=None, detail_level=0):
 
         return object_info(**out)
 
-    def psearch(self,pattern,ns_table,ns_search=[],
-                ignore_case=False,show_all=False):
+    def psearch(self, pattern, ns_table, ns_search=None, ignore_case=False, show_all=False):
```

Comment:
As this is just going to be used for iteration, it might be better to make the default the empty tuple `()` rather than defaulting to `None` and then checking for that in code.